src/main/java/liquibase/nosql/database/AbstractNoSqlDatabase.java

```diff
@@ -0,0 +1,329 @@
+ * %%
+ * Copyright (C) 2020 Mastercard
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import liquibase.CatalogAndSchema;
+import liquibase.database.AbstractJdbcDatabase;
+import liquibase.database.Database;
+import liquibase.database.DatabaseConnection;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LiquibaseException;
+import liquibase.exception.ValidationErrors;
+import liquibase.statement.DatabaseFunction;
+import liquibase.structure.DatabaseObject;
+import lombok.NoArgsConstructor;
+
+import java.math.BigInteger;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * {@link AbstractNoSqlDatabase} is extended by all supported NoSql databases as a facade to the underlying database.
+ * The physical connection can be retrieved from the {@link AbstractNoSqlDatabase} implementation, as well as any
+ * database-specific characteristics.
+ */
+@NoArgsConstructor
+public abstract class AbstractNoSqlDatabase extends AbstractJdbcDatabase implements Database {
+
+    @Override
+    public int getPriority() {
+        return PRIORITY_DATABASE;
+    }
+
+    @Override
+    public boolean supportsInitiallyDeferrableColumns() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsSequences() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsDropTableCascadeConstraints() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsAutoIncrement() {
+        return false;
+    }
+
+    @Override
+    public String getLineComment() {
+        return "//";
+    }
+
+    @Override
+    public String getAutoIncrementClause(final BigInteger startWith, final BigInteger incrementBy, final String generationType, final Boolean defaultOnNull) {
+        return null;
+    }
+
+    @Override
+    public boolean isSystemObject(final DatabaseObject example) {
+        return false;
+    }
+
+    @Override
+    public boolean isLiquibaseObject(final DatabaseObject object) {
+        //TODO: Implement
+        return false;
+    }
+
+    @Override
+    public String getViewDefinition(final CatalogAndSchema schema, final String name) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String escapeObjectName(final String catalogName, final String schemaName, final String objectName, final Class<? extends DatabaseObject> objectType) {
+        return null;
```

Comment:
@filipelautert: what's this supposed to do / does this need to be implemented?
if so: how can this be tested?

(same goes for the other similar methods below)

Reply:
ok, i've removed all the `escape[..]` overrides. though that means that they now return a value instead of `null` and that won't make much sense (but then again, calling the function in the first place didn't make any sense for OpenSearch... so i guess it's ok)